contracts/AirdropCampaignData.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.15;
+
+/// @title AirdropCampaignData - Data contract for storing of daily merkleRootHashes of airbro Campaigns
+contract AirdropCampaignData {
+    address public admin = 0xF4b5bFB92dD4E6D529476bCab28A65bb6B32EFb3;
+    uint16 public batchHashArrayLimit = 600;
+    uint16 public batchChainArrayLimit = 600;
+
+    enum Chains {
+        Zero,
+        Eth,
+        Pols
+    }
+
+    struct AirdropData {
+        Chains chain;
+        bytes32[] hashArray;
+        bool airdropFinished;
+    }
+
+    mapping(address => AirdropData) public airdrops;
+
+    error NotAdmin();
+    error UnequalArrays();
+    error ArrayTooLong();
+    error ChainDataNotSet();
+    error ChainAlreadySet();
+    error AirdropHasFinished();
+    error AlreadyFinalized();
+
+    event AdminChanged(address adminAddress);
+    event MerkleRootHashAdded(address indexed airdropCampaignAddress, bytes32 indexed merkleRootHash);
+    event ChainAdded(address indexed airdropCampaignAddress, Chains indexed airdropChain);
+    event FinalizedAirdrop(address indexed airdropCampaignAddress);
+
+    modifier onlyAdmin() {
+        if (msg.sender != admin) revert NotAdmin();
+        _;
+    }
+
+    constructor() {}
+
+    /// @notice Updates the address of the admin variable
+    /// @param _newAdmin - New address for the admin of this contract, and the address for all newly created airdrop contracts
+    function changeAdmin(address _newAdmin) external onlyAdmin {
+        admin = _newAdmin;
+        emit AdminChanged(_newAdmin);
+    }
+
+    /// @notice Adds daily merkle root hash for any airdropCampaign contract
+    /// @param _airdropCampaignAddress - address of the airdropCampaign contract whose current participants are in the daily merkleRootHash
+    /// @param _merkleRootHash - merkle root hash of daily participants of an airdropCampaign
+    function addDailyMerkleRootHash(address _airdropCampaignAddress, bytes32 _merkleRootHash) external onlyAdmin {
+        if (airdrops[_airdropCampaignAddress].airdropFinished == true) revert AirdropHasFinished();
+        if (airdrops[_airdropCampaignAddress].chain == Chains.Zero) revert ChainDataNotSet();
+
+        airdrops[_airdropCampaignAddress].hashArray.push(_merkleRootHash);
+        emit MerkleRootHashAdded(_airdropCampaignAddress, _merkleRootHash);
+    }
+
+    /// @notice Adds array of daily merkle root hashes for multiple airdropCampaign contracts
+    /// @param _airdropCampaignAddressArray - array of addresses of multiple airdropCampaign contracts whose current participants are in the daily merkleRootHash
+    /// @param _merkleRootHashArray - array of merkle root hashes of daily participants of multiple airdropCampaigns
+    function batchAddDailyMerkleRootHash(address[] memory _airdropCampaignAddressArray, bytes32[] memory _merkleRootHashArray)
```

Comment:
calldata for _airdropCampaignAddressArray

Reply:
Applied it for this and for _merkleRootHashArray 👍